packages/compass-export-to-language/src/stores/store.js

```diff
@@ -88,6 +88,15 @@ const configureStore = (options = {}) => {
       store.dispatch(modalOpenChanged(true));
       store.dispatch(inputExpressionChanged(query));
     });
+
+    localAppRegistry.on(
+      'open-export-to-language-with-mode',
```

Comment:
Feels a bit weird to me that this is not the extension of the `'open-query-export-to-language'` event interface. Don't have the strongest preference, but maybe would make sense to make it one event with the option to pass `mode` as an option?

Reply:
I believe you are right, it would be better to have only one event handler. The only reason I created a new event handler was to avoid changing one that was used already, as I might break some hidden use case.

Reply:
There are only two places in the codebase where this event is emitted, from the usage it seems that you can also simplify the logic there, `typeof string` use-case is just not a thing anymore it seems. If you feel like changing it, please do, this is our code that we maintain, not a third party dependency that we need to work around 🙂 Otherwise we are just accumulating unmaintained code that will get more and more stale